### ISSUE TITLE
Incompatible type on generic field initializer

### ISSUE THREAD

**Issue by eliasku** [TiVo/intellij-haxe#388]
![image](https://cloud.githubusercontent.com/assets/3038174/12576199/0704a7b0-c423-11e5-87be-65154a5afb97.png)

Parsing error (Incompatible type) when short notation for generic instantiation is used. Only for field initializers. 
For example it says `Map<String, String> can't be assigned from Map`
When moving initialization `map = new Map()` to method body - it parses just fine.

Version: 0.9.7 for 14.1.4

Repro:
```
package ;

class Test {

	public var map1:Map<String, String> = new Map<String, String>();
	public var map2:Map<String, String> = new Map();
	
	public var arr1:Array<Int> = [];
	public var arr2:Array<Int> = new Array<Int>();
	public var arr3:Array<Int> = new Array();
	
	public function new() {}
}
```

**Comment by anibyl** [comment]
+1

**Comment by bynuff** [comment]
+1

**Comment by eliasku** [comment]
Please check PR https://github.com/TiVo/intellij-haxe/pull/390

**Comment by EBatTiVo** [comment]
@eliasku - Thanks for the changes.  I merged the PR.  You wouldn't want to add a couple of unit tests, would you?  It would be much appreciated, and would keep us from breaking that functionality again.

**Comment by eliasku** [comment]
FIxed for version 0.9.9
